AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -382,29 +389,39 @@ final class StorageEngine: StorageEngineBehavior {
                 return
             }
         }
-
+        var mutationEventsSubmitCompleted = 0
+        let serialQueue = DispatchQueue(label: "com.amazoncom.StorageEngine.syncDeletions.MType.concurrency")
```

Comment:
Serial queue makes sense to synchronize that whole block, but I don't know that we want to create it in this method. Why not make it a private property of the storage engine so we're not thrashing the system with dispatch queue creations? (We haven't yet done our [audit/refactor of queues](https://github.com/aws-amplify/amplify-ios/issues/139) to set up target queues per subsystem, so you don't need to set targets just yet.)